### PR TITLE
fix segfault

### PR DESCRIPTION
**Pull request by sgaragagghu**

### CODE REVIEW

File: client/block.c [hunk 1]
```diff
@@ -1598,8 +1598,18 @@ int xdag_print_block_info(xdag_hash_t hash, FILE *out)
...
+ 	if(flags & BI_EXTRA) pthread_mutex_lock(&block_mutex);
+
+ 	nlinks = bi->nlinks;
+	memcpy(link, bi->link, nlinks * sizeof(struct block_internal*));
+
+	if(flags & BI_EXTRA) pthread_mutex_unlock(&block_mutex);
+
+ 	for (i = 0; i < nlinks; ++i) {
+		xdag_hash2address(link[i]->hash, address);
 		fprintf(out, "    %6s: %s  %10u.%09u\n", (1 << i & bi->in_mask ? " input" : "output"),
```
Comment (xrdavies) [inline client/block.c]: bi->in_mask will be different, but with extra blocks, it's always "output" for each linked block.

Comment (sgaragagghu) [reply client/block.c]: so ? :joy:

Comment (xrdavies) [reply client/block.c]: It will not have segment issue. Just wrong values.

Comment (sgaragagghu) [reply client/block.c]: ahh yes yes :+1:

**Pull request merged**